lib/mix/lib/mix/tasks/deps.unlock.ex

```diff
@@ -27,6 +30,21 @@ defmodule Mix.Tasks.Deps.Unlock do
       opts[:all] ->
         Mix.Dep.Lock.write(%{})
 
+      opts[:check_unused] ->
+        apps = Mix.Dep.load_on_environment([]) |> Enum.map(& &1.app)
+
+        unused_apps =
+          Mix.Dep.Lock.read()
+          |> Map.drop(apps)
+          |> Map.keys()
+          |> Enum.sort()
+
+        unless unused_apps == [] do
+          Mix.raise(
+            "Unused dependencies in mix.lock file: #{inspect(unused_apps, limit: :infinity)}"
```

Comment:
I think I would prefer better formatting here for the apps. For example:

````
Unused dependencies in mix.lock file:

  * :first_dep
  * :other_dep
```

Reply:
Done.